lib/productions/extended-attributes.js

```diff
@@ -1,8 +1,37 @@
 import { Base } from "./base.js";
 import { ArrayBase } from "./array-base.js";
-import { list, identifiers, argument_list, autoParenter } from "./helpers.js";
+import { Token } from "./token.js";
+import { list, argument_list, autoParenter } from "./helpers.js";
 import { validationError } from "../error.js";
 
+/**
+ * @param {import("../tokeniser").Tokeniser} tokeniser
+ * @param {string} tokenName
+ */
+function tokens(tokeniser, tokenName) {
+  const toks = list(tokeniser, {
+    parser: Token.parser(tokeniser, tokenName),
+    listName: tokenName + " list"
+  });
+  return toks;
+}
+
+/**
+ * This will allow a set of identifiers or strings to be parsed.
```

Comment:
Last nit:

```suggestion
 * This will allow a set of identifiers or strings to be parsed.
 * @param {import("../tokeniser").Tokeniser} tokeniser
```